internal/statesync/metrics.go

```diff
@@ -0,0 +1,91 @@
+package statesync
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/discard"
+	"github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
```

Comment:
A cosmetic note: This appears to be a pre-existing use, so it might not be worth changing. That said: a good rule of thumb is that if we write have a local or third-party package that wraps a larger base project—and re-uses the same name—the _wrapper_ should get renamed rather than the base package.